### PR TITLE
[15.0][MIG] sale_timesheet_line_exclude

### PR DESCRIPTION
**Pull request by CRogos**
#503 has some failing tests. 
There were also a lot of odoo changes between version 13, 14 and 15, so fixing these caused a little bigger refactoring.

### CODE REVIEW

**Comment by CRogos** [comment]
@alexey-pelykh , @elvise , @francesco-ooops  would you also like to review my v15 migration?

**Review by francesco-ooops — APPROVED** [review]
Functional review ok!

**Review by elvise — APPROVED** [review]
LGTM

**Comment by CRogos** [comment]
@alexey-pelykh could you have a look and when merged also close #503

File: sale_timesheet_line_exclude/__manifest__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 {
     "name": "Sales Timesheet: exclude Timesheet Line from Sale Order",
-    "version": "14.0.1.0.0",
+    "version": "15.0.1.0.0",
     "category": "Sales",
     "website": "https://github.com/OCA/timesheet",
     "author": "CorporateHub, " "Odoo Community Association (OCA)",
```
Comment (vincent-hatakeyama) [inline sale_timesheet_line_exclude/__manifest__.py:10]: That line could be cleaned up too by removing `" "`

Comment (vincent-hatakeyama) [inline sale_timesheet_line_exclude/__manifest__.py:10]:
```suggestion
    "author": "CorporateHub, Odoo Community Association (OCA)",
```

That line could be cleaned up a bit.

**Pull request merged**